plugins/module_utils/foreman_helper.py

```diff
@@ -655,10 +690,237 @@ def exit_json(self, changed=False, **kwargs):
+        """ This get current entity and all sub entities based on entity_spec when the param type is 'entity' or 'entity_list'
+             and resource_type is defined. You can also pass 'failsafe' and 'thin' options via entity_spec.
+             Params available for subentities:
+               * resolve (boolean): True by default. If false, this entity will be excluded and not resolved.
+               * failsafe (boolean): False by default.
+               * thin (boolean): True by default.
+               * scope (str): None by default. Provide the scope of the resource (eg: 'organization') (TODO: manage array for 'nested' scope ?)
+                              Defined scope must be a key of entity_spec.
+        """
+        if not entity_dict:
+            entity_dict = self.clean_params()
+
+        self.entity_opts['thin'] = self.desired_absent
+
+        if not entity and self.entity_resolve:
+            if search:
+                entity = self.find_resource(self._entity_resource_path, search,
+                                            failsafe=self.entity_opts['failsafe'],
+                                            thin=self.entity_opts['thin'])
+            else:
+                entity = self.resolve_base_entity(entity_dict)
+
+        if not self.desired_absent:
+            self._resolve_sub_entities(entity, self._unscoped_sub_entities, entity_dict)
+            self._resolve_sub_entities(entity, self._scoped_sub_entities, entity_dict)
+
+            updated_key = "updated_" + self.entity_key
+            if entity and updated_key in entity_dict:
+                entity_dict[self.entity_key] = entity_dict.pop(updated_key)
+
+        return (entity, entity_dict)
+
+    def resolve_base_entity(self, entity_dict):
+        entity = None
+        if 'parent' in self.sub_entities:
+            current, parent = split_fqn(entity_dict[self.entity_key])
+            entity_dict[self.entity_key] = current
+            if 'parent' in entity_dict:
+                if parent:
+                    self.fail_json(msg="Please specify the parent either separately, or as part of the title.")
+                parent = entity_dict['parent']
+
+            if parent:
+                entity_dict['parent'] = self.find_resource_by_title(self._entity_resource_path,
+                                                                    parent, thin=True, failsafe=self.desired_absent)
+                if self.desired_absent and entity_dict['parent'] is None:
+                    # Parent does not exist so just exit here
+                    return entity
+
+            value = build_fqn(entity_dict[self.entity_key], parent)
+        else:
+            value = entity_dict[self.entity_key]
+        # Get current entity with its scope if defined (eg: organization for katello resources)
+        if (self.entity_scope and self.entity_scope in entity_dict and type(entity_dict[self.entity_scope]) is str
```

Comment:
```suggestion
        if (self.entity_scope and self.entity_scope in entity_dict and isinstance(entity_dict[self.entity_scope], str)
```

but also `str` might be wrong here and you get `unicode` or so. Ansible bundles `six`, so you can do `from ansible.module_utils import six` and then `isinstance(entity_dict[self.entity_scope], six.string_types)`